native/src/hooks/useLocation.ts

```diff
@@ -0,0 +1,141 @@
+import Geolocation, { GeolocationError, GeolocationResponse } from '@react-native-community/geolocation'
```

Comment:
I would maybe rename this hook to `useUserLocation` as location is used in many ways.